features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseWireframeMapper.kt

```diff
@@ -31,25 +28,48 @@ abstract class BaseWireframeMapper<T : View, S : MobileSegment.Wireframe>(
     private val viewUtils: ViewUtils = ViewUtils
 ) : WireframeMapper<T, S>, AsyncImageProcessingCallback {
     private var base64Serializer: Base64Serializer? = null
-    private var webPImageCompression: ImageCompression = WebPImageCompression()
-    private var uniqueIdentifierGenerator: UniqueIdentifierGenerator = UniqueIdentifierGenerator
+    private var imageWireframeHelper: ImageWireframeHelper? = null
 
     internal constructor(
-        base64Serializer: Base64Serializer? = null,
-        webPImageCompression: ImageCompression? = null,
-        uniqueIdentifierGenerator: UniqueIdentifierGenerator? = null
+        base64Serializer: Base64Serializer,
+        imageWireframeHelper: ImageWireframeHelper
     ) : this() {
-        base64Serializer?.let {
-            this.base64Serializer = it
-        }
-
-        webPImageCompression?.let {
-            this.webPImageCompression = it
-        }
+        this.base64Serializer = base64Serializer
+        this.imageWireframeHelper = imageWireframeHelper
+    }
 
-        uniqueIdentifierGenerator?.let {
-            this.uniqueIdentifierGenerator = it
+    /**
+     * Maps the [View] into a list of [MobileSegment.Wireframe].
+     */
+    override fun map(view: T, mappingContext: MappingContext): List<S> {
+        val wireframes = mutableListOf<S>()
+
+        val resources = view.resources
+        val density = resources.displayMetrics.density
+        val bounds = resolveViewGlobalBounds(view, density)
+
+        val (shapeStyle, border) = view.background?.resolveShapeStyleAndBorder(view.alpha)
+            ?: (null to null)
+
+        if (view.background != null && shapeStyle == null && border == null) {
+            // try to resolve background as a wireframe
+            // TODO REPLAY-1861 caller thread of .map is unknown?
+            @Suppress( "UNCHECKED_CAST", "ThreadSafety" )
+            imageWireframeHelper?.createImageWireframe(
+                view = view,
+                0,
+                x = bounds.x,
+                y = bounds.y,
+                width = view.width.densityNormalized(density).toLong(),
+                height = view.height.densityNormalized(density).toLong(),
+                drawable = view.background.constantState?.newDrawable(resources),
+                shapeStyle = null,
+                border = null
+            )?.let {
+                wireframes.add(it as S)
```

Comment:
we need to discuss this unsafe cast. Because of type erasure it can't be checked, and this may rule out implementing the background drawable logic in the basewireframemapper

Reply:
The issue is that the unsafe cast can lead to many issues: 
The `ImageWireframeHelper.createImageWireframe()` will always return `MobileSegment.Wireframe.ImageWireframe` when subclasses of `BaseWireframeMapper` might expect different Wireframe types. 
An alternative would be to remove the `S` generic and let map return a generic `MobileSegment.Wireframe` instead. I'm not sure we need to have the exact type of wireframe a in the end it's going to be serialised.

Reply:
I've removed the generic S and am now returning Wireframe